src/components/agent-inbox/utils.ts

```diff
@@ -95,20 +104,56 @@ export function unknownToPrettyDate(input: unknown): string | undefined {
 }
 
 export function constructOpenInStudioURL(
-  deploymentUrl: string,
+  inbox: AgentInbox,
   threadId?: string
-) {
-  const smithStudioURL = new URL("https://smith.langchain.com/studio/thread");
-  // trim the trailing slash from deploymentUrl
-  const trimmedDeploymentUrl = deploymentUrl.replace(/\/$/, "");
+): string {
+  const smithStudioBaseUrl = "https://smith.langchain.com/studio/thread";
 
-  if (threadId) {
-    smithStudioURL.pathname += `/${threadId}`;
-  }
+  if (isDeployedUrl(inbox.deploymentUrl)) {
+    const projectId = extractProjectId(inbox.id);
+    const tenantId = inbox.tenantId;
+
+    if (projectId && tenantId && threadId) {
+      const url = new URL(smithStudioBaseUrl);
+      url.searchParams.set("organizationId", tenantId);
+      url.searchParams.set("hostProjectId", projectId);
+      url.searchParams.set("threadId", threadId);
+      return url.toString();
+    } else {
+      // Handle missing data for deployed graph - return a non-functional link or log error
+      console.warn(
+        "Cannot construct Studio URL for deployed graph: Missing projectId, tenantId, or threadId",
+        {
+          inboxId: inbox.id,
+          hasProjectId: !!projectId,
+          hasTenantId: !!tenantId,
+          hasThreadId: !!threadId,
+        }
+      );
+      return "#"; // Return a placeholder/non-functional link
+    }
+  } else {
+    // --- Existing logic for local/non-deployed URLs ---
+    // TODO: Review if this logic is still appropriate for local development.
+    // Currently appends `baseUrl` and `threadId` to the path.
+    // This might be incorrect if the intention is different for local.
+    const smithStudioURL = new URL(smithStudioBaseUrl);
+    // trim the trailing slash from deploymentUrl
+    const trimmedDeploymentUrl = inbox.deploymentUrl.replace(/\/$/, "");
+
+    if (threadId) {
+      // The original logic appended threadId to the path, which might be wrong.
+      // Let's try adding it as a query param for now, similar to deployed.
+      // Revisit if local URLs need a different structure.
+      // smithStudioURL.pathname += `/${threadId}`; // Original way
```

Comment:
tested locally, what you're doing works so im going to drop these comments
```suggestion
    // --- Logic for local/non-deployed URLs ---
    const smithStudioURL = new URL(smithStudioBaseUrl);
    const trimmedDeploymentUrl = inbox.deploymentUrl.replace(/\/$/, "");

    if (threadId) {
```